YoutubeExplode/YoutubeClient.MediaStream.cs

```diff
@@ -12,49 +12,19 @@ namespace YoutubeExplode
     public partial class YoutubeClient
     {
         /// <inheritdoc />
-        public async Task<MediaStream> GetMediaStreamAsync(MediaStreamInfo info)
+        public Task<MediaStream> GetMediaStreamAsync(MediaStreamInfo info)
         {
             info.GuardNotNull(nameof(info));
+            return Task.FromResult(new MediaStream(info, GetSegmentedStreamForMediaInfo(info)));
+        }
 
-            // Maximum segment stream size
-            const long segmentSize = 9_898_989; // this number was carefully devised through research
-
+        private Stream GetSegmentedStreamForMediaInfo(MediaStreamInfo info)
```

Comment:
No need to make a separate method for this, just inlining is fine.
If you don't want to do `Task.FromResult` you can put `await Task.Yield().ConfigureAwait(false)` before it.

Reply:
I separated it, because I was doing:
```csharp
Stream stream;
if(isRateLimited)
    stream = ...;
else
    stream = ...;
```
and I don't like unassigned variables.
of course now I have code repetition which I also don't like :)

`await Task..` adds more overhead, I prefer doing `Task.FromResult()` where there is no async operations happening, instead of forcing `async` and `await`.